src/utils/thread_helper.hpp

```diff
@@ -258,7 +258,9 @@ class ThreadHelper
 
 #if OTBR_ENABLE_TELEMETRY_DATA_API
     /**
-     * This method populates the telemetry data and returns the error code if error happens.
+     * This method populates the telemetry data with best effort and returns the error code if error
+     * happens. Best effort means the telemetry will be populated if it is available; otherwise it
+     * is left unpopulated.
```

Comment:
From the telemetry analysis's perspective, do we still want the data if the data is incomplete?

Reply:
A last error will be returned. For best effort caller, it can always upload the telemetry as as. For non-best effort caller, the caller can utilize this code to decide whether to upload the telemetry. In my opinion, there are huge amount of telemetries includes, and we want the data as long as the wanted data section is available.